File: Clearance/Models/TemplateCatalog.swift
//
//  TemplateCatalog.swift
//  Clearance
//

import Foundation

enum TemplateFrequency: String, CaseIterable, Identifiable, Sendable {
    case daily, weekly, monthly
    var id: String { rawValue }
    var label: String { rawValue.capitalized }
}

struct TemplateEntry: Identifiable, Sendable {
    let name: String
    let emoji: String
    let activityType: ActivityType
    let frequency: TemplateFrequency
    let tagline: String
    var id: String { name }
}

enum TemplateCatalog {
    static let all: [TemplateEntry] = [
        .init(name: "Gym",     emoji: "🏋️", activityType: .sport, frequency: .weekly, tagline: "Gear-check in, gear-check out"),
        .init(name: "Swim",    emoji: "🏊",  activityType: .sport, frequency: .weekly, tagline: "Pool bag packed the night before"),
        .init(name: "Judo",    emoji: "🥋",  activityType: .sport, frequency: .weekly, tagline: "Gi ready, dojo access confirmed"),
        .init(name: "Cycling", emoji: "🚴",  activityType: .sport, frequency: .weekly, tagline: "Bike check and lights charged"),
        .init(name: "Running", emoji: "🏃",  activityType: .sport, frequency: .weekly, tagline: "Stride pack laid out the night before"),
        .init(name: "Yoga",    emoji: "🧘",  activityType: .sport, frequency: .weekly, tagline: "Mat and props ready to flow"),
        .init(name: "Walking", emoji: "🚶",  activityType: .sport, frequency: .weekly, tagline: "Active recovery — walk and stretch"),
    ]
}

extension ActivityType {
    var label: String {
        switch self {
        case .sport:   "Sport"
        case .work:    "Work"
        case .study:   "Study"
        case .leisure: "Leisure"
        }
    }

    var emoji: String {
        switch self {
        case .sport:   "🏅"
        case .work:    "💼"
        case .study:   "📚"
        case .leisure: "🎮"
        }
    }
}

File: android/app/src/main/java/com/krya1012/clearance/data/TemplateCatalog.kt
package com.krya1012.clearance.data

/** Cadence label shown for a catalog template entry. */
enum class TemplateFrequency {
    DAILY,
    WEEKLY,
    MONTHLY;

    val label: String get() = name.lowercase().replaceFirstChar { it.uppercase() }
}

data class TemplateEntry(
    val name: String,
    val emoji: String,
    val activityType: ActivityType,
    val frequency: TemplateFrequency,
    val tagline: String,
) {
    val id: String get() = name
}

/**
 * Pre-built module templates offered from "Manage modules" so a user can
 * quickly re-add a module they previously deleted (or add one from the
 * catalog on first customization). Mirrors iOS `TemplateCatalog.swift`.
 */
object TemplateCatalog {
    val all: List<TemplateEntry> = listOf(
        TemplateEntry("Gym", "🏋️", ActivityType.SPORT, TemplateFrequency.WEEKLY, "Gear-check in, gear-check out"),
        TemplateEntry("Swim", "🏊", ActivityType.SPORT, TemplateFrequency.WEEKLY, "Pool bag packed the night before"),
        TemplateEntry("Judo", "🥋", ActivityType.SPORT, TemplateFrequency.WEEKLY, "Gi ready, dojo access confirmed"),
        TemplateEntry("Cycling", "🚴", ActivityType.SPORT, TemplateFrequency.WEEKLY, "Bike check and lights charged"),
        TemplateEntry("Running", "🏃", ActivityType.SPORT, TemplateFrequency.WEEKLY, "Stride pack laid out the night before"),
        TemplateEntry("Yoga", "🧘", ActivityType.SPORT, TemplateFrequency.WEEKLY, "Mat and props ready to flow"),
        TemplateEntry("Walking", "🚶", ActivityType.SPORT, TemplateFrequency.WEEKLY, "Active recovery — walk and stretch"),
    )
}
